### DISCUSSION TITLE
Projection on response fields

### DISCUSSION THREAD

**Discussion by thelinuxlich** [ecyrbe/zodios#152] [Q&A] [answered]
How would we do it? The issue is having a `select` array field in the query to choose what fields to return or all if not specified.

**Comment by ecyrbe** [comment, marked as answer]
You can déclare like this
```ts
const userSchema= ...
const endpoint = makeEndpoint({
  method: "get",
  path: "/users",
  parameters: [
    {
       name: "select",
       type: "Query",
       schema: z.array(userSchema.keyof()),
    },
  ],
  response: z.array(userSchema.partial())
});

**Reply by thelinuxlich** [reply]
Hmm interesting, so Zod would need a sophisticated discriminated union and Zodios would need to provide an alternative response as (input) => output like tRPC to truly narrow the response type
